sky/data/data_utils.py

```diff
@@ -908,3 +931,36 @@ def split_oci_path(oci_path: str) -> Tuple[str, str]:
     bucket = path_parts.pop(0)
     key = '/'.join(path_parts)
     return bucket, key
+
+
+def create_coreweave_client(region: Optional[str] = None) -> Client:
+    """Create CoreWeave S3 client.
+    Args:
+        region: The region to configure the client for. If None, uses the
+                region from the session configuration.
+    """
+    session = coreweave.session()
+    if region is not None:
+        return session.client('s3', region_name=region)
+    return session.client('s3')
+
+
+def split_coreweave_path(coreweave_path: str) -> Tuple[str, str]:
+    """Split cw://bucket/key into (bucket, key)."""
+    if coreweave_path.startswith('cw://'):
+        path_parts = coreweave_path.replace('cw://', '').split('/', 1)
+    else:
+        path_parts = coreweave_path.split('/', 1)
+    bucket = path_parts[0]
+    key = path_parts[1] if len(path_parts) > 1 else ''
+    return bucket, key
+
+
+def verify_coreweave_bucket(name: str) -> bool:
+    """Verify CoreWeave bucket exists and is accessible."""
+    coreweave_client = create_coreweave_client()
+    # CoreWeave S3 API doesn't support head_bucket reliably
+    # Use list_buckets instead to check if bucket exists
+    response = coreweave_client.list_buckets()
```

Comment:
We do, just not immediately after creation (we're improving it soon). 

You can either wait for the update and then just use HeadBucket, or you can perform a DNS lookup, and as soon as you return any responses with a `CNAME` record then HeadBucket will work reliably.

Reply:
Using DNS lookup